File: kotlin-multiplatform/monstradore/androidApp/src/main/java/com/example/monstradore/android/performance/PerformanceView.kt
package com.example.monstradore.android.performance

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.material.Text
import androidx.compose.material.TextField
import androidx.compose.material3.Button
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.monstradore.structures.prime
import kotlinx.coroutines.*

@OptIn(DelicateCoroutinesApi::class)
@Composable
fun PerformanceContent() {

    var input by remember { mutableStateOf("20000") }
    var result by remember { mutableStateOf(0) }
    var time by remember { mutableStateOf(0L) }
    Column (
        modifier = Modifier.padding(10.dp)
    ) {
        Text(
            text = "Performance",
            fontSize = 25.sp,
            fontWeight = FontWeight.Bold
        )
        Text(text = "Die wievielte Primzahl soll berechnet werden?")
            TextField(
                value = input,
                onValueChange = { input = it },
                label = { Text("Zahl eingeben") }
            )
            Button(onClick = {
                GlobalScope.launch(Dispatchers.IO) {
                    val starttime = System.currentTimeMillis()
                    val primeResult = prime(input.toInt())
                    val endtime = System.currentTimeMillis()
                    time = (endtime - starttime) / 1000
                    withContext(Dispatchers.Main) {
                        result = primeResult
                    }
                }
            }) {
                Text("Berechnen")
        }
        Text("Ergebnis: $result")
        Text("Benötigte Zeit: $time Sekunden")
    }
}

File: kotlin-multiplatform/monstradore/iosApp/iosApp/performance/PerformanceView.swift
//
//  PerformanceView.swift
//  iosApp
//
//  Created by Anja on 29.11.22.
//  Copyright © 2022 orgName. All rights reserved.
//

import SwiftUI
import shared

private func prime(n: Int32) async -> Int32 {
    PerformanceKt.prime(n: n)
}

struct PerformanceView: View {
    @State private var primeValue = ""
    @State private var speed = 0.0
    @State private var isEditing = false
    @State private var result = Int32(0)
    @State private var time = 0
    var body: some View {
        VStack {
            Text("Performance")
                .font(.headline)
            Text("Die wievielte Primzahl soll berechnet werden?")
            TextField("Zahl eingeben", text: $primeValue)
            Button(action: {
                let startTime = Int64(NSDate().timeIntervalSince1970 * 1000)
                Task {
                    result = await prime(n: Int32(primeValue) ?? 0)
                    let endTime = Int64(NSDate().timeIntervalSince1970 * 1000)
                    time = Int(endTime - startTime) / 1000
                }
            }) {
                Text("Berechnen")
            }
            Text("Ergebnis: \(result)")
            Text("Benötigte Zeit: \(time) Sekunden")
            Slider(
                value: $speed,
                in: 0...100,
                onEditingChanged: { editing in
                    isEditing = editing
                }
            )
        }
        .padding()
    }
}

//struct PerformanceView_Previews: PreviewProvider {
//    static var previews: some View {
//        PerformanceView()
//    }
//}
